src/icrar/leap-accelerate/MetaData.cc

```diff
@@ -143,22 +143,30 @@ namespace icrar
      * @param metadata 
      * @param direction 
      */
-    void SetDD(MetaData& metadata, const MVDirection& direction)
+    void MetaData::SetDD(const MVDirection& direction)
     {
-        metadata.dlm_ra = direction.get()[0] - metadata.phase_centre_ra_rad;
-        metadata.dlm_dec = direction.get()[1] - metadata.phase_centre_dec_rad;
+        const int I2D = 2;
 
-        metadata.dd(IPosition(0,0)) = cos(metadata.dlm_ra) * cos(metadata.dlm_dec);
-        metadata.dd(IPosition(0,1)) = -sin(metadata.dlm_ra);
-        metadata.dd(IPosition(0,2)) = cos(metadata.dlm_ra) * sin(metadata.dlm_dec);
+        if(!dd.is_initialized())
+        {
+            dd.reset(casacore::Matrix<double>(3,3));
+        }
+
+        auto& dd3d = dd.value();
+        dlm_ra = direction.get()[0] - phase_centre_ra_rad;
+        dlm_dec = direction.get()[1] - phase_centre_dec_rad;
+
+        dd3d(IPosition(I2D,0,0)) = cos(dlm_ra) * cos(dlm_dec);
+        dd3d(IPosition(I2D,0,1)) = -sin(dlm_ra);
+        dd3d(IPosition(I2D,0,2)) = cos(dlm_ra) * sin(dlm_dec);
         
-        metadata.dd(IPosition(1,0)) = sin(metadata.dlm_ra) * cos(metadata.dlm_dec);
-        metadata.dd(IPosition(1,1)) = cos(metadata.dlm_ra);
-        metadata.dd(IPosition(1,2)) = sin(metadata.dlm_ra) * sin(metadata.dlm_dec);
+        dd3d(IPosition(I2D,1,0)) = sin(dlm_ra) * cos(dlm_dec);
+        dd3d(IPosition(I2D,1,1)) = cos(dlm_ra);
+        dd3d(IPosition(I2D,1,2)) = sin(dlm_ra) * sin(dlm_dec);
 
-        metadata.dd(IPosition(2,0)) = -sin(metadata.dlm_dec);
-        metadata.dd(IPosition(2,1)) = 0;
-        metadata.dd(IPosition(2,2)) = cos(metadata.dlm_dec);
+        dd3d(IPosition(I2D,2,0)) = -sin(dlm_dec);
+        dd3d(IPosition(I2D,2,1)) = 0;
+        dd3d(IPosition(I2D,2,2)) = cos(dlm_dec);
```

Comment:
`casacore::Matrix` defines an `T &operator()(size_t i1, size_t i2)`, so you should be able to do a more direct indexing in all these assignments; e.g. `dd3d(0, 0) = cos(dlm_ra) * cos(dlm_dec);`